### ISSUE TITLE
Java14 Full Records Support Check Validation: EqualsAvoidNullCheck

### ISSUE THREAD

**Issue by nmancus1** [checkstyle/checkstyle#8464]
Child of #8452 
Check documentation: https://checkstyle.sourceforge.io/config_coding.html#EqualsAvoidNull

From check  documentation: 
Checks that any combination of String literals is on the left side of an equals() comparison. Also checks for String literals assigned to some field (such as someString.equals(anotherString = "text")).

```
➜  full-record-grammar /usr/lib/jvm/java-14-openjdk/bin/javac --enable-preview --source 14 TestClass.java
Note: TestClass.java uses preview language features.
Note: Recompile with -Xlint:preview for details.
➜  full-record-grammar cat config.xml                                                                    
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
      <module name="EqualsAvoidNull"/>
  </module>
</module>
➜  full-record-grammar cat TestClass.java
public class TestClass {
    public record MyRecord1() {
        public boolean stringCheck1 (String toCheck) {
            return toCheck.equals("my string"); // violation
        }

        public boolean stringCheck2 (String toCheck) {
            return "my string".equals(toCheck); // ok
        }
    }
}
➜  full-record-grammar java $RUN_LOCALE -jar ~/IdeaProjects/checkstyle/target/checkstyle-8.35-SNAPSHOT-all.jar -c config.xml TestClass.java
Starting audit...
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:4:34: String literal expressions should be on the left side of an equals comparison. [EqualsAvoidNull]
Audit done.
Checkstyle ends with 1 errors.

```
This check works as intended, i think we can close this issue.

**Comment by pbludov** [comment]
This check should be updated to handle compact ctors:
```java
class TestClass {
    String str;
    public TestClass() {
        str.equals("");  // violation
    }
}

record TestRecord(String str) {
    public TestRecord {
        str.equals(""); // expected violation
    }
}
```

```
Starting audit...
[ERROR] /home/pbludov/src/tmp/TestClass.java:4:19: String literal expressions should be on the left side of an equals comparison. [EqualsAvoidNull]
Audit done.
```

**Comment by romani** [comment]
Fix is merged
